chat/rich-links/src/RichLink/RichLink.types.ts

```diff
@@ -0,0 +1,55 @@
+import { DarkModeProps, HTMLElementProps } from '@leafygreen-ui/lib';
+
+import { Variant as RichLinkBadgeVariantName } from './RichLinkBadge/RichLinkBadge.types';
+import { RichLinkVariantName } from './RichLinkVariants';
+
+export interface BaseRichLinkProps
+  extends HTMLElementProps<'a', HTMLAnchorElement>,
+    DarkModeProps {
+  /**
+   * The URL that the rich link points to
+   */
+  url: string;
```

Comment:
Does this component only support external links?

Reply:
For our use case we'll be point to URLs that may be on the docs, dev center, youtube, etc

Was there an alternative you were thinking of here?

Reply:
I'm not sure what other teams might use this component, but there is a chance that if someone wants to use an internal link and they're using nextjs, we would need to support the `as` prop.

Reply:
Good point! I'm not too familiar with as/polymorphism but I can take a look at some other components and let you know if I have any questions on how to implement.

Reply:
The [Link](https://github.com/mongodb/leafygreen-ui/tree/main/packages/typography/src/Link/Link) component might be helpful

Reply:
Took some time to look into this but it's not quite the same use case as Link and TS is misbehaving with the Card `as` prop which complicates things further. Given that we won't be using this on the docs chatbot, can we leave polymorphism as a future feature?

Reply:
I pulled the `rich-links-polymorphic` [branch](https://github.com/mongodb/leafygreen-ui/blob/rich-links-polymorphic/chat/rich-links/src/RichLink/RichLink.tsx), and I'm not running into any build errors. I'm honestly not sure why it's giving you issues. 

Also, `Card` is tricky because it uses `Box`. We haven't gotten around to updating it to use `Polymorphic`, so I think it's okay to ignore the errors.

@bruugey Do you think we should proceed without polymorphic links since this is causing a few issues?